### PR TITLE
Use the privileged mode to exec on the container

### PR DESCRIPTION
**Pull request by lubosmj**
This allows us to run podman pull/push commands inside the container. Also, the change is required to run pulp_container tests in general.

[noissue]

### CODE REVIEW

File: client/oci_env/utils.py [hunk 1]
```diff
@@ -383,9 +383,9 @@ def exec(self, args, service=None, interactive=False, pipe_output=False):
         # docker fails on systems with no interactive CLI. This tells docker
         # to use a pseudo terminal when no CLI is available.
         if os.getenv("COMPOSE_INTERACTIVE_NO_CLI", "0") == "1":
-            cmd = [binary, "exec", self.container_name(service)] + args
+            cmd = [binary, "exec", "--privileged", self.container_name(service)] + args
```
Comment (mdellweg) [inline client/oci_env/utils.py:386]: I'd prefer to find a way without this.

https://docs.podman.io/en/latest/markdown/podman-exec.1.html#privileged

Comment (lubosmj) [reply client/oci_env/utils.py:386]: `binary` can also be a docker client. Not sure if it is worth doing any complicated things.

ref: https://www.redhat.com/sysadmin/podman-inside-container

Comment (decko) [reply client/oci_env/utils.py:386]: What if we add a `--privileged` flag to oci-env where you explicitly choose to run the container like that?

Comment (lubosmj) [reply client/oci_env/utils.py:386]: Fair, any objections, @mdellweg?

Comment (mdellweg) [reply client/oci_env/utils.py:386]: Sounds ok. Just don't break security by default for convenience.

**Comment by lubosmj** [comment]
This does not address the problem of running podman commands via `oci-env test -i - p`. Not sure why. I can run podman calls from within the container without problems though.

**Comment by decko** [comment]
> This does not address the problem of running podman commands via `oci-env test -i - p`. Not sure why. I can run podman calls from within the container without problems though.

@lubosmj Do we have an issue describing this issue?

**Review by decko — APPROVED** [review]
LGTM

**Pull request merged**